martha/martha_v3.js

```diff
@@ -478,87 +489,130 @@ async function retrieveFromServers(params) {
         `and access method type '${accessMethodType}'`
     );
 
-    let response;
-    if (overlapFields(requestedFields, MARTHA_V3_METADATA_FIELDS)) {
+    let bondSA;
+    let drsResponse;
+    let fileName;
+    let accessUrl;
+
+    // `fetch` below might time out in a way that we want to report as an error. Since the timeout
+    // interrupts us while we're waiting for a response, we need to capture what we were about to
+    // try doing just before we do it so that we can provide that detail in the error report.
+    let hypotheticalErrorMessage;
+
+    const fetch = async () => {
```

Comment:
Best viewed ignoring whitespace. The changes here are:
1. Wrapping all of the server fetching into a `fetch` function
2. Moving the SA key fetch from the bottom to the top
3. Pulling all of the `let`s out of the function so they can be accessed after calling the function
4. Managing `hypotheticalErrorMessage` (🤢 )
5. Wrapping everything involved in fetching a signed URL in a `try`/`catch` to log instead of throwing